### PR TITLE
This should allow RequestHeader statements to be passed to a vhost.

### PR DESCRIPTION
**Pull request by Aethylred**
This. can either be a single string or and array of statements.

More detail on RequestHeader declarations can be found here:
http://httpd.apache.org/docs/2.2/mod/mod_headers.html#requestheader

### CODE REVIEW

File: manifests/default_mods.pp [hunk 1]
```diff
@@ -60,4 +59,5 @@
   apache::mod { 'authz_host': }
   apache::mod { 'authz_user': }
   apache::mod { 'env': }
+  apache::mod { 'headers': }
```
Comment (hunner) [inline manifests/default_mods.pp]: I believe the `headers` module is not enabled by default on Debian, which is why it was not included here. I may be wrong though; which version of Debian were you testing this on?

Comment (hunner) [inline manifests/default_mods.pp]: Perhaps we could create an `apache::mod::headers` class than can conditionally be included inside `apache::vhost` when `request_header` is passed.

File: templates/vhost/_requestheader.erb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+# as per http://httpd.apache.org/docs/2.2/mod/mod_headers.html#requestheader
+  <% if @request_header.is_a? Array -%>
+  <% @request_header.each do |request_statement| -%>
+  RequestHeader <%= request_statement %>
+  <% end -%>
+  <% elsif @request_header != '' -%>
+  RequestHeader <%= @request_header %>
+  <% end -%>
+
+<% end -%>
```
Comment (hunner) [inline templates/vhost/_requestheader.erb:14]: You can also ensure `@request_headers` is a non-emptystring array no matter what is passed in like:

``` erb
<% if @request_headers and ! @request_headers.empty? -%>

  ## Request rules
<% Array(@request_headers).each do |request_header| -%>
    RequestHeader <%= request_header %>
<% end -%>
<% end -%>
```

Also remember to update the `apache::vhost` template comment with the new fragment and used variables.

File: manifests/vhost.pp [hunk 3]
```diff
@@ -86,6 +86,7 @@
     $redirect_dest      = undef,
     $redirect_status    = undef,
     $rack_base_uris     = undef,
+    $request_header     = undef,
```
Comment (hunner) [inline manifests/vhost.pp]: Perhaps this should be `$request_headers` since it could accept an array?

File: manifests/default_mods.pp (hunk 1 above)
Comment (Aethylred) [inline manifests/default_mods.pp]: I use Ubuntu 12.10, which has `mod_headers` on by default, but I like the suggestion of enabling `apache::mod::headers` (which doesn't exist yet) automatically if `request_headers` is passed.

File: manifests/vhost.pp (hunk 3 above)
Comment (Aethylred) [inline manifests/vhost.pp]: Agreed.

File: templates/vhost/_requestheader.erb (hunk 2 above)
Comment (Aethylred) [inline templates/vhost/_requestheader.erb:14]: Yes, and should also do nothing if the array contains empty strings `['','']`

...actually the strings should be validated, but I'm not writing a syntax for it now.

**Pull request closed without merging**